Heat_method_3/include/CGAL/Heat_method_3/Surface_mesh_geodesic_distances_3.h

```diff
@@ -374,36 +374,36 @@ class Surface_mesh_geodesic_distances_3
       Index i = get(vertex_id_map, current);
       Index j = get(vertex_id_map, neighbor_one);
       Index k = get(vertex_id_map, neighbor_two);
-      VertexPointMap_reference p_i = get(vpm,current);
+      VertexPointMap_reference p_i = get(vpm, current);
       VertexPointMap_reference p_j = get(vpm, neighbor_one);
       VertexPointMap_reference p_k = get(vpm, neighbor_two);
       Index face_i = get(face_id_map, f);
 
-      Vector_3 v_ij = construct_vector(p_i,p_j);
-      Vector_3 v_ik = construct_vector(p_i,p_k);
-      Vector_3 cross = cross_product(v_ij, v_ik);
-      double norm_cross = CGAL::sqrt(to_double(scalar_product(cross,cross)));
-      double dot = to_double(scalar_product(v_ij, v_ik));
-      double cotan_i = dot/norm_cross;
-
-      Vector_3 v_ji = construct_vector(p_j, p_i);
-      Vector_3 v_jk = construct_vector(p_j, p_k);
-
-      cross = cross_product(v_ji, v_jk);
-      dot = to_double(scalar_product(v_ji, v_jk));
-      double cotan_j = dot/norm_cross;
-
-      Vector_3 v_ki = construct_vector(p_k,p_i);
-      Vector_3 v_kj = construct_vector(p_k,p_j);
-
-      cross = cross_product(v_ki, v_kj);
-      dot = to_double(scalar_product(v_ki,v_kj));
-      double cotan_k = dot/norm_cross;
-
-      const Vector_3& a  = m_X[face_i];
-      double i_entry = (to_double(scalar_product(a,v_ij)) * cotan_k) + (to_double(scalar_product(a,v_ik)) *  cotan_j);
-      double j_entry = (to_double(scalar_product(a,v_jk)) * cotan_i) + (to_double(scalar_product(a,v_ji)) * cotan_k);
-      double k_entry = (to_double(scalar_product(a,v_ki)) * cotan_j) + (to_double(scalar_product(a,v_kj)) * cotan_i);
+      const Vector_3 v_ij = construct_vector(p_i, p_j);
+      const Vector_3 v_ik = construct_vector(p_i, p_k);
+      const Vector_3 v_ji = construct_vector(p_j, p_i);
+      const Vector_3 v_jk = construct_vector(p_j, p_k);
+      const Vector_3 v_ki = construct_vector(p_k, p_i);
+      const Vector_3 v_kj = construct_vector(p_k, p_j);
+
+      const Traits traits;
+      const FT cotan_i = CGAL::Weights::
+        cotangent(p_k, p_i, p_j, traits);
```

Comment:
This carriage return is completely unmotivated, that is the line is not too long and the cr makes it harder to read.

Reply:
True. Fixed.